examples/async-reply-hook/src/wasm.rs

```diff
@@ -0,0 +1,79 @@
+// This file is part of Gear.
+
+// Copyright (C) 2023-2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use gstd::{critical, debug, exec, msg, prelude::*, ActorId};
+
+#[gstd::async_main]
+async fn main() {
+    let source = msg::source();
+
+    // Case 1: Message without reply hook
+    let m1 = gstd::msg::send_bytes_for_reply(source, b"for_reply_1", 0, 0)
+        .expect("Failed to send message");
+
+    // Case 2: Message with reply hook but we don't reply to it
+    let m2 = gstd::msg::send_bytes_for_reply(source, b"for_reply_2", 0, 1_000_000_000)
+        .expect("Failed to send message")
+        .up_to(Some(5))
+        .expect("Failed to set timeout")
+        .handle_reply(|| {
+            unreachable!("This should not be called");
```

Comment:
@breathx do we have auto-replies for messages sent to user mailbox? Or they work just for app-2-app interactions?

Reply:
answered in person - we do have ones

Reply:
According to the answer, the message here has to be amended (or comment should be added) to something like - "This should be called in gas / 100 blocks, but the program exits by that time"